soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -2968,7 +2968,12 @@ void GenerateRandomizerImgui(std::string seed = "") {
     cvarSettings[RSK_GERUDO_KEYS] = CVarGetInteger("gRandomizeGerudoKeys", RO_GERUDO_KEYS_VANILLA);
     cvarSettings[RSK_KEYRINGS] = CVarGetInteger("gRandomizeShuffleKeyRings", RO_KEYRINGS_OFF);
     cvarSettings[RSK_KEYRINGS_RANDOM_COUNT] = CVarGetInteger("gRandomizeShuffleKeyRingsRandomCount", 8);
```

Comment:
Setting count to 9, and then turning fast carpenters on after, then generating a seed would result in 9 being passed in, but the resulting `keyRingOptions` would have a size of 8 (out of bounds).
Need to do the max count check and probably use std::min here.
```suggestion
    // Handle correct key ring count, accounting for fortress ring being in the pool or not
    int maxKeyringCount = (CVarGetInteger("gRandomizeGerudoFortress", RO_GF_NORMAL) == RO_GF_NORMAL &&
                           CVarGetInteger("gRandomizeGerudoKeys", RO_GERUDO_KEYS_VANILLA) != RO_GERUDO_KEYS_VANILLA)
                                ? 9
                                : 8;
    cvarSettings[RSK_KEYRINGS_RANDOM_COUNT] = std::min(CVarGetInteger("gRandomizeShuffleKeyRingsRandomCount", maxKeyringCount), maxKeyringCount);
```

Reply:
Have you tested that? I'm pretty sure I tried that and it passed in 8. I'll double check.

Reply:
OK, I must've tested with changing the Gerudo Keys setting rather than the Carpenters setting, I'll get a fix for that shortly.

Reply:
Fixed